main.go

```diff
@@ -31,6 +31,13 @@ var version string
 
 func main() {
 
+	product := lib.GetProductById(parameters.Product)
+	if product == nil {
+		fmt.Printf("Invalid product specified")
+		lib.UsageMessage()
+		os.Exit(0)
```

Comment:
It's probably a matter of personal preference, though I think it is odd to exit successfully on invalid user input.

Reply:
Yes, I wasn't quite sure how to handle this...
`GetParameters` already obtains the product and ensures it's not null. Ideally this should be done once - but I'm not sure if GetParameters or main should do do it

Reply:
> guess, ideally, `GetParameters` would store the Product that it obtains and main wouldn't need to check

+1 👍🏻

Reply:
I have added the product entity to Params in 67433dbaf7d6fb94c7f51486edb0a2b0cf1f659e and removed this code ;)